### ISSUE TITLE
Duplicate underlying datasource ID in TS Data Connections

### ISSUE THREAD

**Issue by MarnixKolder** [tableau/community-tableau-server-insights#38]
In the example below I have two data published sources, one on one site, and one on another.
There's two workbooks on each site and both contain an embedded data source referencing their site's respective published data source.
However you can see below [TS Data Connections] reports two underlying data sources for each workbook - one is the correct one, but one is the corresponding data source from the other site.

![TS Data Connections](https://user-images.githubusercontent.com/63275724/80782608-35bd5580-8bba-11ea-86b7-885e8e38f7a2.png)

Is this a bug? Can it be rectified?
I am trying to count the number of workbooks connected to each data source, but I'm getting the wrong information.
If I was to count the workbooks connecting to data source 3774, I'd get 4 - two on the Default site and two on the Development site. Now this has to be wrong, of course, because we know we can't connect a workbook on one site to a published data source on another.

Thanks

**Comment by MarnixKolder** [comment]
Further to this, I had a go at at least creating a temporary solution.

I basically have limited understanding of what I am doing, but I altered the name [Data Sources (Custom SQL)] table to include the following clause:
![image](https://user-images.githubusercontent.com/63275724/80785405-82595e80-8bc3-11ea-83ae-54c87555001b.png)

Then I just use this new field to link to a copy of the [sites] table to pull info about the site related to the underlying datasource:
![image](https://user-images.githubusercontent.com/63275724/80785360-59d16480-8bc3-11ea-915f-16e351e437ab.png)

Bringing the underlying site name into view allows me to see where the Site Name and Underlying Site Name agree or not.
![image](https://user-images.githubusercontent.com/63275724/80785535-f8f65c00-8bc3-11ea-805e-3cf4a708c6c2.png)

I can then filter out where they disagree and I get the correct results - 2 published data sources, one on each site with two workbooks connected to each of those two datasources...
![image](https://user-images.githubusercontent.com/63275724/80785602-2e02ae80-8bc4-11ea-8fb3-f40383017c55.png)

As I say, it's a temporary fix and hardly rigorous, but it might help in creating something more permanent...

**Comment by mcoles** [comment]
Pretty sure it's this join that is the problem, in the Data Sources (Custom SQL):

![image](https://user-images.githubusercontent.com/14928335/80944884-64485580-8d9f-11ea-952a-120b927b6860.png)

**Comment by MarnixKolder** [comment]
Is that because the link is based on a name and I have assets with the same name on different sites?

**Comment by mcoles** [comment]
Yes, exactly. At the time I originally built this, I had been under the impression that the repository_urls were unique across server. But that doesn't actually make sense, and was never true. So this is a bug, and we just need to add the criteria of site_id into the join.

**Comment by donnacoles30** [comment]
Just came on here to advise I'd found issue, and then seen this thread - doh!

Yeah I fixed by adding an additional join query to the join statement Matt has highlighted above : p_dc.site_id = p_ds.site_id

**Comment by mcoles** [comment]
This is resolved with the 04.01 release.
